qiskit/_result.py

```diff
@@ -414,3 +414,22 @@ def get_qubitpol_vs_xval(self, nqubits, xvals_dict=None):
                     circuit_name, z_dicts[qubit_ind])
 
         return qubitpol, xvals
+
+
+def copy_qasm_from_qobj_into_result(qobj, result):
+    """Find the several QASM belonging to the Qobj experiments and copy them
+    into the corresponding result entries."""
+    for experiment in qobj.experiments:
+        name = experiment.header.name
+        qasm = getattr(experiment.header, 'compiled_circuit_qasm', None)
+        if qasm:
+            experiment_result = _find_experiment_result(result, name)
+            experiment_result['compiled_circuit_qasm'] = qasm
+
+
+def _find_experiment_result(result, name):
+    for experiment_result in result['result']:
+        if experiment_result['name'] == name:
+            return experiment_result
+
+    return None
```

Comment:
This one I'm a bit more concerned about: is there really a case where it could really return `None`? If that is the case, line 427 would fail at:
```
experiment_result['compiled_circuit_qasm'] = qasm
```

Reply:
As far as I know, nope, it would expose an inconsistency between the `Qobj` and `Result` instances but the linter complained. I can add a warning and avoid accessing it if `None` is returned. Does it make sense?

Reply:
Yeah I think explicitly taking care of the `None` case would be good.

Reply:
I'm going to keep this as a private function. With the one-liner, I can not add the warning and by moving it inside the `copy_qasm_from_qobj_into_result` function, makes the function hard to read.